packages/cel/src/equals.ts

```diff
@@ -0,0 +1,222 @@
+// Copyright 2024-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { isMessage } from "@bufbuild/protobuf";
+import {
+  CelList,
+  CelMap,
+  CelObject,
+  CelType,
+  CelUint,
+  ProtoNull,
+} from "./value/value.js";
+import {
+  isReflectList,
+  isReflectMap,
+  isReflectMessage,
+  reflect,
+  type ReflectList,
+  type ReflectMap,
+} from "@bufbuild/protobuf/reflect";
+import { getEvalContext } from "./eval.js";
+import { equals as equalsMessage } from "@bufbuild/protobuf";
+import {
+  DurationSchema,
+  isWrapper,
+  TimestampSchema,
+} from "@bufbuild/protobuf/wkt";
+
+/**
+ * Checks for equality of two CEL values. It follows the following rules:
+ *
+ * - Primitives are checked for value equality using native equality check.
+ * - Types need not be equal for numerical values to be equal.
+ * - Wrappers are treated as scalars.
```

Comment:
"native equality check" is ambiguous because ECMAScript has two:
1. the equality operator `==`
2. the strict equality operator `===`

I think we're more clear if we use CEL's type names here. What about:

```suggestion
 * - Numeric values (`int`, `uint`, and `double`) are compared across types. 
 * - `NaN` is not equal `NaN`.
 * - Other scalar values (`bool`, `string`, `bytes`, and `type`) are only equal if type and value are identical. 
 * - Wrapped scalars (e.g. `google.protobuf.StringValue`) are unwrapped before comparison.
 * - `google.protobuf.Any` is unpacked before comparison.
```

Can you add a link to https://github.com/google/cel-spec/blob/v0.24.0/doc/langdef.md#equality ?